tests/graph/test_opt.py

```diff
@@ -630,8 +625,9 @@ def test_patternsub_values_eq_approx(out_pattern, tracks):
         assert output is x
         assert output.tag.values_eq_approx is values_eq_approx_always_true
     else:
-        assert isinstance(output, Constant)
-        assert not hasattr(output.tag, "value_eq_approx")
+        # The replacement types do not match, so the substitution should've
```

Comment:
I think I put this condition because it was suported by the PatternSub according to the docstrings

https://github.com/aesara-devs/aesara/blob/9fc8113c8f87bd4735b32413c8c467fcc13b7a84/aesara/graph/opt.py#L1578

Reply:
The pattern still accepts them; now, they're converted to `Variable`s when the pattern is parsed.

The current/old approach would let those "native" numeric types pass through the optimization framework unconverted.  In this particular case, that wasn't a problem, because, when `var.type.filter_variable` is eventually called, it would fall back on `MyType`'s inherited `Type.filter_variable` and produce a raw `Constant`.

The new approach forces the types to be explicit upfront, which should help avoid confusion and unexpected errors.